hyper/http20/connection.py

```diff
@@ -522,6 +523,16 @@ def _consume_single_frame(self):
         # Parse the header. We can use the returned memoryview directly here.
         frame, length = Frame.parse_frame_header(header)
 
+        frame_size = self._settings[SettingsFrame.SETTINGS_MAX_FRAME_SIZE]
+        if (length > frame_size):
+            self._send_rst_frame(frame.stream_id, 6) # 6 = FRAME_SIZE_ERROR
+            log.warning(
+                "Frame size exceeded on stream %d (received: %d, max: %d)",
+                frame.stream_id,
+                length,
+                frame_size
+            )
```

Comment:
It'd be safer to log first, I think: that way the log is emitted before we explode due to any traceback.